backends-clickhouse/src/main/scala/org/apache/gluten/backendsapi/clickhouse/CHIteratorApi.scala

```diff
@@ -55,18 +55,24 @@ import scala.collection.mutable.ArrayBuffer
 
 class CHIteratorApi extends IteratorApi with Logging with LogLevelUtil {
 
-  private def getFileSchema(schema: StructType, names: Seq[String]): StructType = {
-    val dataSchema = ArrayBuffer[StructField]()
+  private def getFileSchema(schema: StructType, names: Seq[String]): java.util.List[SchemaField] = {
```

Comment:
Can we just create a class type that holds multiple field elements? Then, directly replace StructType with the new class type.